Misc/NEWS.d/next/Library/2025-08-26-09-07-02.gh-issue-135763.DVpGUB.rst

```diff
@@ -0,0 +1,2 @@
+Add ``allow_negative`` flag to ``Py_ssize_t`` Argument Clinic converter, to
```

Comment:
AC is unfortunately an internal tool so I don't think we should document it. **However**, what could be interesting is whether we want to expose a new parsing character for positive `Py_ssize_t` values. For now, https://docs.python.org/3/c-api/arg.html#numbers only documents `n` for `Py_ssize_t` but maybe having something for `Py_ssize_t >= 0` would make sense (I don't know, so for now, don't change anything).

Reply:
I agree no NEWS entry is needed, but @wiomoc could create a PR for the devguide after this is merged, which is where we document AC.

Reply:
> I wonder why not providing a format unit for conversion to size_t directly.

Because we use ssize_t for most internal sizes in Python as we need to indicate when a conversion failed with value = -1 && PyErr_Occurred() (I think).

>  The only advantage is see with Py_ssize_t(allow_negative) in comparison to size_t is the fact that you could provide a negative value e.g. -1 default value as undefined / None sentinel.

Unfortunately, internally, most structures use `ssize_t` for sizes and not `size_t`, so we could have overflows here (that is, we pass a value that fits on a size_t but not on a ssize_t).